common/unsafe/src/main/java/org/apache/spark/sql/catalyst/util/CollationFactory.java

```diff
@@ -1113,6 +1113,13 @@ public CollationIdentifier identifier() {
   public static final List<String> SUPPORTED_PROVIDERS = List.of(PROVIDER_SPARK, PROVIDER_ICU);
   public static final String COLLATION_PAD_ATTRIBUTE = "NO_PAD";
 
+  public static final String UTF8_BINARY_COLLATION_NAME =
```

Comment:
How about to add new class `CollationNames` with those constants to `org.apache.spark.sql.catalyst.util`:
```java
public class CollationNames {
  public static final String UTF8_BINARY = ...
```

BTW, the suffix `_COLLATION_NAME` could be eliminated I believe.